### PR TITLE
Added the `http.get` and `http.post` functions with query params, headers, form, bodies as optional params (depending on GET vs POST)

### PR DESCRIPTION
**Pull request by Cictrone**
#### What type of PR is this?

/kind feature
/kind eldritch-function

#### What this PR does / why we need it:
It will be useful to issue HTTP GET requests from imix.

### CODE REVIEW

File: implants/lib/eldritch/src/http/get_impl.rs [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+    }
+
+    runtime.block_on(handle_get(full_uri, headers_map))
+}
+
+fn append_query_params_to_uri(
+    mut uri: String,
+    query_params: SmallMap<String, String>,
+) -> Result<String> {
+    let mut after_first_param = false;
```
Comment (KCarretto) [inline implants/lib/eldritch/src/http/get_impl.rs:35]: Just use hyper

File: implants/lib/eldritch/src/http/post_impl.rs [hunk 2]
```diff
@@ -0,0 +1,228 @@
+use anyhow::Result;
+use reqwest::header::{HeaderMap, HeaderName, HeaderValue};
```
Comment (KCarretto) [inline implants/lib/eldritch/src/http/post_impl.rs:2]: why are we using reqwest?

File: implants/lib/eldritch/src/http/post_impl.rs [hunk 3]
```diff
@@ -0,0 +1,228 @@
...
+) -> Result<String> {
+    #[cfg(debug_assertions)]
+    log::info!(
+        "eldritch sending HTTP POST request to '{}' with headers '{:#?}'",
+        uri,
+        headers
+    );
+
+    let client = reqwest::Client::new().post(uri).headers(headers);
+    if body.is_some() {
```
Comment (KCarretto) [inline implants/lib/eldritch/src/http/post_impl.rs:55]: `if let Some(b) = body`

File: implants/lib/eldritch/src/http/post_impl.rs [hunk 4]
```diff
@@ -0,0 +1,228 @@
...
+        let resp = client.body(body.unwrap()).send().await?.text().await?;
+        return Ok(resp);
+    }
+    if form.is_some() {
```
Comment (KCarretto) [inline implants/lib/eldritch/src/http/post_impl.rs:59]: `if let Some(f) = form`

File: implants/lib/eldritch/src/http/post_impl.rs (hunk 2 above)
Comment (Cictrone) [reply implants/lib/eldritch/src/http/post_impl.rs:2]: science

File: implants/lib/eldritch/src/http/get_impl.rs (hunk 1 above)
Comment (Cictrone) [reply implants/lib/eldritch/src/http/get_impl.rs:35]: no

**Review by KCarretto — APPROVED** [review]

**Pull request merged**